Signal/src/network/GiphyDownloader.swift

```diff
@@ -639,13 +726,18 @@ extension URLSessionTask {
                     continue
                 }
 
-                guard assetRequest.firstActiveSegment() == nil else {
-                    activeAssetRequestsCount += 1
-                    // Ensure that only N requests are active at a time.
-                    guard activeAssetRequestsCount < self.kMaxAssetRequestCount else {
-                        return nil
-                    }
-
+                let downloadingSegmentsCount = assetRequest.downloadingSegmentsCount()
+                activeAssetRequestsCount += downloadingSegmentsCount
+                // Ensure that only N segment requests are active per asset at a time.
+                guard downloadingSegmentsCount < kMaxAssetRequestsPerAssetCount else {
```

Comment:
Naming suggests this should be `<=`

Reply:
See above, we want to enforce the limit and not exceed it.